### PR TITLE
feat: add sliding panels to data explorer for facets, resources, and preview (#441)

### LINKED ISSUE #441: [TASK] Data explorer - Expandable cart panel
Enhance the current cart container/panel to have expanded and collapsed states

In collapsed mode

Image

In expanded mode

Image

### PR DESCRIPTION
**Pull request by mj3cheun**
# Description

Adds a general sliding panel component ("Slider.vue") and changes the facet, resources, and preview panels to use it. A few changes were made to some of the other data explorer components (e.g. Document) to make it work with the new slider component as well as a few general fixes.

Screen Shot 2023-01-16 at 3 02 56 PM

Resolves #441

### CODE REVIEW

**Comment by YohannParis** [comment]
Please add a description and assign someone to the PR. 😄

File: packages/client/hmi-client/src/views/DataExplorer.vue [hunk 1]
```diff
@@ -1,13 +1,26 @@
...
-				:facets="facets"
-				:filtered-facets="filteredFacets"
-				:result-type="resultType"
-			/>
+			<slider content-width="20%" tab-width="48px" direction="left" :is-open="isSliderFacetsOpen">
+				<template v-slot:content>
+					<div class="slider-header content">
+						<IconChevronLeft24 class="slider-header-item" @click="isSliderFacetsOpen = false" />
+						<h4 class="slider-header-item">Facets</h4>
+					</div>
```
Comment (YohannParis) [inline packages/client/hmi-client/src/views/DataExplorer.vue:9]: This seems something that we might use a lot.
Could you make it a slot `header` like in the modal component?

File: packages/client/hmi-client/src/views/DataExplorer.vue [hunk 2]
```diff
@@ -1,13 +1,26 @@
...
+						:facets="facets"
+						:filtered-facets="filteredFacets"
+						:result-type="resultType"
+					/>
+				</template>
+				<template v-slot:tab>
+					<div class="slider-header tab">
+						<IconChevronRight24 class="slider-header-item" @click="isSliderFacetsOpen = true" />
+					</div>
+				</template>
```
Comment (YohannParis) [inline packages/client/hmi-client/src/views/DataExplorer.vue:22]: I think this should be handle by the `slider` component as well.
Using the prop direction of opening, we can infer those icons.

File: packages/client/hmi-client/src/components/Slider.vue [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+	>
+		<div class="slider-content-container" :style="{ width: contentWidth }">
+			<div class="slider-content" :style="sidePanelContentStyle">
+				<slot name="content"></slot>
+			</div>
+		</div>
+		<div class="slider-tab" :style="sidePanelTabStyle">
+			<slot name="tab"></slot>
+		</div>
+	</div>
```
Comment (YohannParis) [inline packages/client/hmi-client/src/components/Slider.vue:14]: So many divs... you want my eyes to bleed 😭 
Could we use `<main>`, `<header>` etc,

**Comment by mwdchang** [comment]
Can we call this component something else than slider? Maybe "SlidingPanel" "CollapsePanel" or something? 

When people are talking Slider in the UI context they are typically referring to this

image

**Comment by mj3cheun** [comment]
im open to an and all suggestions and sliding panel sounds good. side note, at some point we should also have a conversation about why we got sidebar, sidepanel, and sidebar-panel components lol. the chosen name was partially to try to avoid adding another name to the list.

File: packages/client/hmi-client/src/components/data-explorer/slider-panel.vue [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+			<IconChevronRight24 class="slider-header-item" @click="emit('update:isOpen', true)" />
+			<h5 class="slider-header-item">{{ header }}</h5>
+			<slot name="tab"></slot>
+		</template>
+	</slider>
+</template>
+
+<script setup lang="ts">
+import IconChevronLeft24 from '@carbon/icons-vue/es/chevron--left/24';
+import IconChevronRight24 from '@carbon/icons-vue/es/chevron--right/24';
```
Comment (YohannParis) [inline packages/client/hmi-client/src/components/data-explorer/slider-panel.vue:25]: Could you use [PrimeVue icons](https://www.primefaces.org/primevue/icons)?

File: packages/client/hmi-client/src/components/data-explorer/slider-panel.vue [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+// const directionMap = {
+// 	left: {
+// 		content: 'margin-left: -100%; margin-right: 100%;',
+// 		tab: 'margin-left: auto;'
+// 	},
+// 	right: {
+// 		content: 'margin-left: 100%; margin-right: -100%;',
+// 		tab: 'margin-right: auto;'
+// 	}
+// };
```
Comment (YohannParis) [inline packages/client/hmi-client/src/components/data-explorer/slider-panel.vue:66]: You can remove this.

**Review by YohannParis — APPROVED** [review]

**Review by YohannParis — APPROVED** [review]

**Pull request merged**